libyul/AsmPrinter.cpp

```diff
@@ -245,11 +245,11 @@ std::string AsmPrinter::formatTypedName(TypedName _variable)
 
 std::string AsmPrinter::appendTypeName(YulString _type, bool _isBoolLiteral) const
 {
-	if (m_dialect && !_type.empty())
+	if (m_printingMode == Mode::OmitDefaultType && !_type.empty())
```

Comment:
Would be best to assert that `m_printingMode` is either this or `Mode::FullTypeInfo` so that adding a new value to the enum and not handling it produces an error.